accounts/tests.py

```diff
@@ -1575,3 +1576,166 @@ def test_bulk_describe_state_closed(self):
         self.assertIn('This bulk description process is closed', resp.content)
+
+
+class ReportSpamOffensive(TestCase):
+    """
+    Test the "report spam/offensive" feature available for sound comments, forum posts and private messages.
+    NOTE: for simplicity in  variable names, etc we only refer to the "spam" case.
+    """
+
+    fixtures = ['initial_data', 'sounds']
+
+    def setUp(self):
+        # Create users reporting spam
+        self.reporters = []
+        for i in range(0, settings.USERFLAG_THRESHOLD_FOR_AUTOMATIC_BLOCKING + 1):
+            reporter = User.objects.create_user(username='reporter_{0}'.format(i),
+                                                email='reporter_{0}@example.com'.format(i), password='testpass')
+            self.reporters.append(reporter)
+
+        # Create user posting spam
+        self.spammer = User.objects.create_user(username='spammer', email='spammer@example.com', password='testpass')
+
+    def get_reporter_as_logged_in_user(self, i):
+        user = self.reporters[i]
+        self.client.force_login(user=user)
+        return user
+
+    def __test_report_object(self, flag_type, object):
+
+        # Flag the comment (no email to admins should be sent yet as only one reporter)
+        reporter = self.get_reporter_as_logged_in_user(0)
+        resp = self.client.post(reverse('flag-user', kwargs={'username': self.spammer.username}), data={
+            'object_id': object.id,
+            'flag_type': flag_type,
+        })
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(UserFlag.objects.count(), 1)  # One flag object created
+        self.assertEqual(UserFlag.objects.first().reporting_user, reporter)  # Flag object created by reporter
+        self.assertEqual(len(mail.outbox), 0)  # No email sent
+
+        # Flag the same comment by other users so email is sent
+        for i in range(1, settings.USERFLAG_THRESHOLD_FOR_NOTIFICATION):  # Start at 1 as first flag already done
+            reporter = self.get_reporter_as_logged_in_user(i)
+            resp = self.client.post(reverse('flag-user', kwargs={'username': self.spammer.username}), data={
+                'object_id': object.id,
+                'flag_type': flag_type,
+            })
+            self.assertEqual(resp.status_code, 200)
+            self.assertEqual(UserFlag.objects.count(), i + 1)  # Now we have more flags
+            self.assertEqual(UserFlag.objects.all().order_by('id')[i].reporting_user,
+                             reporter)  # Flag object created by reporter
+
+            if i == settings.USERFLAG_THRESHOLD_FOR_NOTIFICATION - 1:  # Last iteration
+                self.assertEqual(len(mail.outbox), 1)  # Notification email sent
+                self.assertTrue("[freesound] Spam/offensive report for user" in mail.outbox[0].subject)
+                self.assertTrue("has been reported" in mail.outbox[0].body)
```

Comment:
If this patch was to include the objects in the email, should we check for them here?

Reply:
Yes we could but it would introduce some "if else" to get the content from the object and check if it's in the email body which would be a bit ugly. I think it is not really needed as if there is a problem retrieving the content the test will fail, and in any case when these emails are sent to the admins we will quickly identify the potential issue. I'd argue that for the sake of simplicity there is no need to add this checks.